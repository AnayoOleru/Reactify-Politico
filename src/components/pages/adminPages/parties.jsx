import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import swal from 'sweetalert';
import { getAllParties } from '../../../actions/getActions';
import AdminNavBar from '../../reuseable component/admin-navbar.component.jsx';
import '../../../styles/parties.style.css';

export class Parties extends Component {
  componentDidMount() {
    const { getAllParties } = this.props;
    getAllParties();
  }

  handleDeleteParty() {
    swal({
      title: "Are you sure?",
      icon: "warning",
      button: true,
      dangerMode: true,
      text: "cancel",
    }).then((willDelete) => {
      if (willDelete) {
        swal("Poof! Your imaginary file has been deleted!", {
          icon: "success",
        });
      } else {
        swal("Your imaginary file is safe!");
      }
    });
  }
  render() {
    const style5 = {
      fontSize: '30px'
    };

    const style4 = {
      backgroundColor: '#ffffff',
    };

    const focus = {
      backgroundColor: '#ffffff',
    };

    const focusColor = {
      color: '#0b4156',
    };

    const getParties = this.props.get && this.props.get.map(party => (
      <div className="col-1-of-3" key={party.id}>
        <div className="card">
          <div className="card__side card__side--front">
            <img className="card__picture card__picture--3" src={party.logourl} />
            <div className="card__details">
              <ul>
                <li style={style5}>{party.name}</li>
              </ul>
            </div>
          </div>
          <div className="card__side card__side--back card__side--back-3">
            <div className="card__cta">
              <div className="card__price-box">
                <p className="card__price-only">Headquater Address</p>
                <p className="card__price-only">{party.hqaddress}</p>

              </div>
              <Link to={`/parties/${party.id}/name`}>
                <a href="#" className="btn" id="edit">Edit</a>
              </Link>
              <Link to={`/parties/${party.id}/delete`}>
                <a href="#" className="btn" id="delete">Delete</a>
              </Link>
            </div>
          </div>
        </div>
      </div>
    ));
    return (
      <React.Fragment>
        <AdminNavBar props={this.props} focus={focus} focusColor={focusColor}  />
        <main style={style4}>
          <section className="section-cards">
            <div className="text-cards">
              <h1 className="heading-primary">
                Parties
                </h1>
              <div className="row" id="partyResult">{getParties}</div>
            </div>
            <div className="row" id="partyResult" />
            <Link to={'/add-party'}><button href={'/add-party'} className="add" >+</button></Link>

          </section>
        </main>
      </React.Fragment>
    );
  }
}
Parties.propTypes = {
  getAllParties: PropTypes.func.isRequired,
};


const mapStateToProps = state => ({
  posts: state.payload,
  get: state.get.items.data,
});

export default connect(mapStateToProps, { getAllParties })(Parties);
