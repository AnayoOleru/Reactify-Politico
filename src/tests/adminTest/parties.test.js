import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { Parties } from '../../components/pages/adminPages/parties.jsx';

describe('Parties component should', () => {
    const props = {
        getAllParties: jest.fn(),
    };
    const wrapper = shallow(<Parties {...props} />);
  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
