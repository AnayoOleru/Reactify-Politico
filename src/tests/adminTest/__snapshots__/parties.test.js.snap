// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parties component should renders without crashing 1`] = `ShallowWrapper {}`;
